File: AspNetMvcUnleashed/0672329980_source/Chapter11/CS/MvcApplication1/MvcApplication1/Models/MovieRepositoryBase.cs
using System.Collections.Generic;

namespace MvcApplication1.Models
{
    public abstract class MovieRepositoryBase
    {
        public abstract IEnumerable<Movie> ListMoviesCached();
        public abstract IEnumerable<Movie> ListMovies();
        public abstract void CreateMovie(Movie movieToCreate);
    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter11/VB/MvcApplication1/MvcApplication1/Models/MovieRepositoryBase.vb
Public MustInherit Class MovieRepositoryBase

    Public MustOverride Function ListMoviesCached() As IEnumerable(Of Movie)
    Public MustOverride Function ListMovies() As IEnumerable(Of Movie)
    Public MustOverride Sub CreateMovie(ByVal movieToCreate As Movie)

End Class
